ui/src/main/java/com/doordeck/sdk/ui/verify/VerifyDevicePresenter.kt

```diff
@@ -67,11 +49,35 @@ internal class VerifyDevicePresenter {
         jobs += GlobalScope.launch(Dispatchers.Main) {
 
             val ephKey = ImmutableRegisterEphemeralKey.builder().ephemeralKey(Doordeck.getKeys().public).build()
-            val result: Response<Void> = client!!.certificateService().initVerification(ephKey, method).awaitResponse()
+            val result: Response<Map<String, String>> = client!!.certificateService().initVerification(ephKey).awaitResponse()
             when (result.isSuccessful) {
                 true -> {
+                    val usedMethod = result.body()?.get("method");
```

Comment:
Null check is good, but you can just make the returned value a VerificationMethod enum by calling VerificationMethod.valueOf(usedMethod) to save yourself all the string comparisons below

Reply:
fixed in 2c05f41